packages/vc-verification/src/claim-issuer-verification.ts

```diff
@@ -136,4 +159,39 @@ export class ClaimIssuerVerification {
     }
     return claim instanceof Object;
   }
+
+  /**
+   * Checks the revocation status for the given StatusList2021Entry and role
+   * @param statusListEntry credentialStatus associated with the VC
+   * @param role namespace
+   * @returns
+   */
+  async checkRevocationStatus(
```

Comment:
I think there is some duplication with this `checkRevocationStatus` method and the `RevocationVerification` class. I think instead of this method, this `ClaimIssuerVerification` class should instantiate a `RevocationVerification` instance. In that way, our revocation check logic isn't duplicated. 

The same is the case in `VcIssuerVerification`

Does this make sense to you @nichonien @JGiter ?

Reply:
Yes, revocation verification is more appropriate for this

Reply:
Yeah, I did that initially. But then I realised there is a circular dependency as `RevocationVerification` takes `ClaimIssuerVerification` and `VCIssuerVerification`. So instantiating `RevocationVerification` in `IssuerVerification` didn't seem the right way. Or am I missing something @jrhender @JGiter ?

Reply:
Hmm, I don't think there is a circular dependency issue 🤞 . Did you already try it out and it failed? It's going to be challenging to do it in `ClaimIssuerVerification` and `VCIssuerVerification` but in `IssuerVerification`, I think it is clearer how it is possible.

This is the current `IssuerVerification` constructor:
```
constructor(
    private issuerResolver: IssuerResolver,
    private credentialResolver: CredentialResolver,
    vcIssuerVerification: VCIssuerVerification,
    claimIssuerverification: ClaimIssuerVerification
  ) {
    this.vcIssuerVerification = vcIssuerVerification;
    this.claimIssuerverification = claimIssuerverification;
  }
```
I think one could just add another property
```
constructor(
    private issuerResolver: IssuerResolver,
    revokerResolver: RevokerResolver,
    private credentialResolver: CredentialResolver,
    vcIssuerVerification: VCIssuerVerification,
    claimIssuerverification: ClaimIssuerVerification
  ) {
    this.vcIssuerVerification = vcIssuerVerification;
    this.claimIssuerverification = claimIssuerverification;
    this.revocationVerification = new RevocationVerification(revokerResolver, credentialResolver, vcIssuerVerification, claimIssuerVerification)
  }
```

Regarding `ClaimIssuerVerification` and `VCIssuerVerification`, I think we can perhaps make `claimIssuerVerification` and `vcIssuerVerification` optional parameters to `RevocationVerification` and then `ClaimIssuerVerification` and `VCIssuerVerification` can just provide the one they have. For example,

```
constructor(
    private issuerResolver: IssuerResolver,
    private revocationResolver: RevocationResolver,
    private credentialResolver: CredentialResolver,
    private verifyProof: (vc: string, proof_options: string) => Promise<any>
  ) {
   this.revocationVerification = new RevocationVerification(revokerResolver, credentialResolver, claimIssuerVerification: this)
}
```

Reply:
@jrhender If we do that in `IssuerVerification`, `IssuerVerification` calls respective `ClaimIssuerVerification` and `VCIssuerVerification` for verification of the issuer chain based on the credential fetch for the given issuer. Now, if we want to verify the revocation status of each credential in the chain then it needs to call the revocation verification code internally and thus again the same approach. But let me try it once.

Reply:
>But let me try it once.

@nichonien Ok, thanks. let me know how it goes